lib/ansiblelint/rules/MetaMainHasInfoRule.py

```diff
@@ -4,16 +4,49 @@
 from ansiblelint.rules import AnsibleLintRule
 
 
+META_STR_INFO = [
+    'author',
+    'description',
+]
+META_INFO = META_STR_INFO + [
+    'license',
+    'min_ansible_version',
+    'platforms',
+]
+
+
+def _platform_info_errors_itr(platforms):
+    if not isinstance(platforms, list):
+        yield 'Platforms should be a list of dictionaries'
+        return
+
+    for platform in platforms:
+        if not isinstance(platform, dict):
+            yield 'Platforms should be a list of dictionaries'
+        elif 'name' not in platform:
+            yield 'Platform should contain name'
+
+
+def _galaxy_info_errors_itr(galaxy_info,
+                            info_list=META_INFO,
+                            str_info_list=META_STR_INFO):
+    for info in info_list:
+        ginfo = galaxy_info.get(info, False)
+        if ginfo:
```

Comment:
you can easily dedent the whole block of code by just investing this check to be a guard expression:
```suggestion
        if not ginfo:
            yield 'Role info should contain 
            continue
```

Reply:
Thanks for reviewing and your comment!

> you can easily dedent the whole block of code by just investing this check to be a guard expression:

I cannot remember its exact reason but maybe I prefer to avoid explicit call of 'continue' and the negating conditional statement.

Anyway, I don't have strong opinions about this and if it's one of requirements for merge, I'll change that.

Reply:
I usually prefer guard-expressions because they simplify the code improving the readability. But I guess we can keep this for now.